pkg/api/datasources.go

```diff
@@ -385,10 +386,22 @@ func (hs *HTTPServer) CheckDatasourceHealth(c *models.ReqContext) {
 		return
 	}
 
+	var jsonDetails map[string]interface{}
 	payload := map[string]interface{}{
 		"status":      resp.Status.String(),
 		"message":     resp.Message,
-		"jsonDetails": resp.JSONDetails,
+		"jsonDetails": jsonDetails,
+	}
+
+	// Unmarshal JSONDetails if its not empty.
+	if len(resp.JSONDetails) > 1 {
```

Comment:
Should it be the following? If the length is 1 it's not empty.
```suggestion
	if len(resp.JSONDetails) > 0 {
```

Reply:
Needs at least `{}` to be valid JSON.

Reply:
But maybe your suggestion is cleaner. If the plugin returns `{` we want to return an error.

Reply:
@bergquist Yeah, I think we might as well uniformly return an error on any invalid JSON, it would also harmonize with the comment that says "if it's not empty".

Reply:
agree!